### PR TITLE
reworks emote sound code & adds screaming (Which only works if the code forces it. *scream wont make a sound)

### PR DESCRIPTION
**Pull request by ghost**
:cl: Floyd / Qustinnus
soundadd: involuntary screaming for when you get hurt.
code: reworks emote sound handling to not use fucking s u b t y p i n g
/:cl:

so basically I like sounds and screaming adds some important feelies so I made it so that if the code forces you to emote a scream, it will make you scream audibly with audio. but *scream doesnt work (to prevent people ruining it like they did with every other fun feature)

https://streamable.com/yfksw

### CODE REVIEW

**Comment by nicbn** [comment]
Make it louder please

**Comment by Qustinnus** [comment]
why itll be obnoxious

**Comment by Qustinnus** [comment]
also recording is a bit softer than the real thing for most of my PRs for some reason

**Comment by nicbn** [comment]
Currently it's too quiet - I couldn't even hear the first female scream at the first time.
Also the scream should only happen in involuntary situations so it won't be that bad if it's louder.
Compare the scream to the shutter noise: the shutter is much louder. I think a good volume for the scream would be a bit louder than the shutter.

**Comment by optimumtact** [comment]
@nicbn the first scream is produced when the person is slammed by the door and appears adequately balanced to me

**Comment by Qustinnus** [comment]
@nicbn i think you missed the point, the first scream didnt make a sound because it was voluntary, which is mentioned in the PR and a comment for a total count of 3 times 🤔

**Comment by Ratvar** [comment]
time to beat mutadoned monkeys to generate screams

**Comment by Vidiveni** [comment]
Test this with styptic powder smoke.

**Comment by GuyonBroadway** [comment]
Coming from hippie manual screaming actually adds a new dimension of play that makes you a little less helpless.

Say its the age old story of a ling stuns you and you get cuffed. Before he can remove your radio you just have time to scream "LING CHAPEL MAINT" before he drags you into a side room. Sec do a sweep of the area while he silently gives you the succ. No matter how hard you yell "HELP IN THE ELECTRICAL ROOM" even a guy two tiles away, even if you could hear the lings backpack rustle through a wall at that range your bloody yelling goes unheard and you get succ'd before sec find the room you are in.

But if you can *scream to literally scream for help then sec have a chance at finding you before you get rendered the very ded. On hippie manual screaming causes oxyloss so spamming it is difficult thus using it in a situation like that is a tricky tactical choice. 

Of course it won't save you if sec is incompetent(hint: they are) but having some tiny mechanism of resistance makes you dying feel that much more engaging.

File: code/datums/emotes.dm [hunk 1]
```diff
@@ -59,6 +63,12 @@
 	user.log_message(msg, LOG_EMOTE)
 	msg = "<b>[user]</b> " + msg
 
+	if(get_sound(user))
+		if(!only_forced_audio)
+			playsound(user.loc, get_sound(user), 50, vary)
```
Comment (Cyberboss) [inline code/datums/emotes.dm]:
```suggestion
			playsound(user, get_sound(user), 50, vary)
```

File: code/datums/emotes.dm [hunk 2]
```diff
@@ -59,6 +63,12 @@
 	user.log_message(msg, LOG_EMOTE)
 	msg = "<b>[user]</b> " + msg
 
+	if(get_sound(user))
```
Comment (Cyberboss) [inline code/datums/emotes.dm]: Store the result in a variable and don't call it twice

File: code/datums/emotes.dm [hunk 3]
```diff
@@ -59,6 +63,12 @@
...
+		if(!only_forced_audio)
+			playsound(user.loc, get_sound(user), 50, vary)
+		else if(!intentional)
```
Comment (Cyberboss) [inline code/datums/emotes.dm]:
```suggestion
```

File: code/datums/emotes.dm [hunk 4]
```diff
@@ -59,6 +63,12 @@
...
+			playsound(user.loc, get_sound(user), 50, vary)
```
Comment (Cyberboss) [inline code/datums/emotes.dm]:
```suggestion
```

File: code/datums/emotes.dm [hunk 5]
```diff
@@ -59,6 +63,12 @@
 	user.log_message(msg, LOG_EMOTE)
 	msg = "<b>[user]</b> " + msg
 
+	if(get_sound(user))
+		if(!only_forced_audio)
```
Comment (Cyberboss) [inline code/datums/emotes.dm]:
```suggestion
		if(!only_forced_audio || !intentional)
```

File: code/modules/mob/living/carbon/human/emote.dm [hunk 6]
```diff
@@ -45,6 +45,29 @@
...
+	emote_type = EMOTE_AUDIBLE
+	only_forced_audio = TRUE
+	vary = TRUE
+
+/datum/emote/living/carbon/human/scream/get_sound(mob/user)
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if((!H.mind || !H.mind.miming))
+			if(ishumanbasic(H) || iscatperson(H))
+				if(user.gender == FEMALE)
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/emote.dm]: Early return, my dude

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Swiftkor** [comment]
Is there some kind of cooldown between screams? Is there a damage threshold for screaming?
Also, you should probably make sure this doesn't apply to mimes.

**Comment by zxaber** [comment]
If you ask me, the first sample played in the video sounds a lot more like "oh, spoopy ghost" scream rather than "ow, my arm broke" scream. The second male scream is rather long as well.

Otherwise, neat idea. I look forward to medbay becoming the station's choir.

File: code/modules/mob/living/carbon/human/emote.dm (hunk 6 above)
Comment (Qustinnus) [reply code/modules/mob/living/carbon/human/emote.dm]: ?

**Comment by nicbn** [comment]
@Qustinnus I meant the first involuntary scream

**Comment by Qustinnus** [comment]
how did you not hear that lol

**Comment by improvedname** [comment]
@guyonBroadway >not having mute sting as ling

File: code/modules/mob/living/carbon/human/emote.dm (hunk 6 above)
Comment (Cyberboss) [reply code/modules/mob/living/carbon/human/emote.dm]:
```dm
if(!ishuman(user))
    return
var/mob/living/carbon/human/H = user
if(!H.mind?.miming)
    return
```

Comment (Qustinnus) [reply code/modules/mob/living/carbon/human/emote.dm]: okay fixed

File: code/datums/emotes.dm [hunk 7]
```diff
@@ -22,6 +22,10 @@
 	var/stat_allowed = CONSCIOUS
 	var/static/list/emote_list = list()
 
+	var/sound //Sound to play when emote is called
```
Comment (Cyberboss) [inline code/datums/emotes.dm]: Move static stuff to the bottom

File: code/datums/emotes.dm [hunk 8]
```diff
@@ -59,6 +63,11 @@
 	user.log_message(msg, LOG_EMOTE)
 	msg = "<b>[user]</b> " + msg
 
+	var/tmp_sound = get_sound(user)
+	if(tmp_sound)
+		if(!only_forced_audio || !intentional)
```
Comment (Cyberboss) [inline code/datums/emotes.dm]: Combine ifs

File: code/datums/emotes.dm [hunk 9]
```diff
@@ -71,6 +80,9 @@
 	else
 		user.visible_message(msg)
 
+/datum/emote/proc/get_sound(mob/user)
```
Comment (Cyberboss) [inline code/datums/emotes.dm]: If ghosts can't emote change to mob/living

File: code/modules/mob/living/carbon/alien/emote.dm [hunk 10]
```diff
@@ -12,19 +12,18 @@
 	message_alien = "hisses."
 	message_larva = "hisses softly."
 
-/datum/emote/living/alien/hiss/run_emote(mob/user, params)
-	. = ..()
-	if(. && isalienadult(user))
-		playsound(user.loc, "hiss", 40, 1, 1)
+/datum/emote/living/alien/hiss/get_sound(mob/user)
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/alien/emote.dm]: ^

File: code/modules/mob/living/carbon/alien/emote.dm [hunk 11]
```diff
@@ -12,19 +12,18 @@
...
 	message_alien = "roars."
 	message_larva = "softly roars."
 	emote_type = EMOTE_AUDIBLE
+	vary = TRUE
 
-/datum/emote/living/alien/roar/run_emote(mob/user, params)
-	. = ..()
-	if(. && isalienadult(user))
-		playsound(user.loc, 'sound/voice/hiss5.ogg', 40, 1, 1)
+/datum/emote/living/alien/roar/get_sound(mob/user)
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/alien/emote.dm]: ^

File: code/modules/mob/living/carbon/human/emote.dm [hunk 12]
```diff
@@ -45,6 +45,31 @@
...
 
+/datum/emote/living/carbon/human/scream
+	key = "scream"
+	key_third_person = "screams"
+	message = "screams!"
+	emote_type = EMOTE_AUDIBLE
+	only_forced_audio = TRUE
+	vary = TRUE
+
+/datum/emote/living/carbon/human/scream/get_sound(mob/user)
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/emote.dm]: ^

File: code/modules/mob/living/emote.dm [hunk 13]
```diff
@@ -204,22 +204,22 @@
...
 /datum/emote/living/laugh/can_run_emote(mob/living/user, status_check = TRUE)
 	. = ..()
 	if(. && iscarbon(user))
 		var/mob/living/carbon/C = user
 		return !C.silent
 
-/datum/emote/living/laugh/run_emote(mob/user, params)
-	. = ..()
-	if(. && ishuman(user))
+/datum/emote/living/laugh/get_sound(mob/user)
```
Comment (Cyberboss) [inline code/modules/mob/living/emote.dm]: ^

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Qustinnus** [comment]
fuck i didnt mean to push the tool thing smh

**Comment by tortellinitony** [comment]
Separate PR can enable *scream if people want it

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**